src/Chapter2.hs

```diff
@@ -858,9 +870,9 @@ and reverses it.
   function, but in this task, you need to implement it manually. No
   cheating!
 -}
-rewind = error "rewind: Not Implemented!"
-
-
+rewind :: [a] -> [a]
+rewind [] = []
+rewind x = last x : rewind (init x)
```

Comment:
Your solution is correct! However, it is slow. In lists, it is quite slow to add anything at the end of the list or get elements from the end. That is why it is always better to rewrite it with the `:` cons and access only the head and tail of the list. Remember the explanation with trains? 🚂 🚋 🚋 

That is why a more efficient solution is with the accumulator and the recursive function that will do the addition at the start of the list which is instant!

You can read a bit more about the `go` pattern in here: https://kowainik.github.io/posts/haskell-mini-patterns#recursive-go